woocommerce/google-listings-and-ads
# Add a script to generate a list of hooks that defined or used in GLA

### Changes proposed in this Pull Request:

This PR adds a script to generate a list of hooks (actions or filters) that is defined or used in this project. The output file is generated in `src/Hooks/README.md`.

The majority of the codes were copied fro [WooCommerce Code Reference](https://github.com/woocommerce/code-reference/blob/331da252e94b3e6529d2d42f6636200ca30dcfe4/generate-hook-docs.php) repo, and they're published in https://woocommerce.github.io/code-reference/hooks/hooks.html.

Since the hooks in GLA project are mostly used internally so generated as a markdown file would be good enough.

### Detailed test instructions:

1. Run `php bin/HooksDocsGenerator.php`
2. See if the script generated a markdown file in `src/Hooks/README.md`

### Changelog entry

> Add - A script to generate a list of hooks that defined or used in GLA.
